Add some requested CLI cookbook entries

BlaineHeffron opened the pull request:
Addresses https://github.com/stellar/stellar-cli/issues/1438 

Adds cookbook entries for:

- A lifecycle of a contract (requested by the community)
- Payment for XLM, classic asset, soroban assets

We cannot yet do the following because the commands are not available on the CLI:
- Issuance for Stellar and Soroban assets  (specifically we need trustline functionality)
- Transaction (should mostly be available once this PR is merged https://github.com/stellar/stellar-cli/pull/1406

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr840.previews.kube001.services.stellar-ops.com

Ifropc commented on docs/build/guides/cli/contract-lifecycle.mdx:
````diff
@@ -0,0 +1,52 @@
+1. Create an identity for Alice:
+
+```bash
+stellar keys generate alice
````
Do you mind adding here `--no-fund` as a recommended flag to use? We are planning to make `--no-fund` to be a [default behavior in 22.0](https://github.com/stellar/stellar-cli/issues/1407) so it'd be nice if people were aware of this change (or just noting that it won't be auto-funded in future versions is fine)

BlaineHeffron replied:
I added --no-fund and an additional step of funding. Once it becomes default we can remove the --no-fund part from this doc.

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr840.previews.kube001.services.stellar-ops.com

Ifropc commented on docs/build/guides/cli/payments-and-assets.mdx:
````diff
@@ -0,0 +1,42 @@
+1. Fund the accounts:
+
+```bash
+stellar keys generate alice --network testnet
````
`--no-fund`?

Ifropc commented on docs/build/guides/cli/payments-and-assets.mdx:
```diff
@@ -0,0 +1,42 @@
+
+To send payments and work with assets using the Stellar CLI, follow these steps:
+
+1. Fund the accounts:
```
WDYT about `export STELLAR_NETWORK=testnet` as the first step in the guide? Would shorten commands a bit

Ifropc commented on docs/build/guides/cli/contract-lifecycle.mdx:
````diff
@@ -0,0 +1,59 @@
+3. Deploy a contract:
+
+```bash
+stellar contract deploy --wasm /path/to/contract.wasm --source alice --network testnet
````
Do you mind pointing to the docs that explain how to generate WASM here?

BlaineHeffron replied:
Put a link to the getting started tutorial which seemed like a good place to point people to https://developers.stellar.org/docs/build/smart-contracts/getting-started/setup

Ifropc commented:
LGTM, just added a couple of small comments

janewang commented:
@BlaineHeffron Ideally this could be in a single markdown file called cookbooks.mdx, with different paths for each recipe.

janewang commented:
@BlaineHeffron I was thinking that this Cookbook.mdx could be added to the CLI repo next to the  [FULL_HELP_DOCS.md](https://github.com/stellar/stellar-cli/blob/main/FULL_HELP_DOCS.md) and we could copy it over the docs, similar to the CLI Manual is copied over: https://github.com/stellar/stellar-docs/pull/755

BlaineHeffron commented:
@janewang Makes sense, I'll work on getting that set up.

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr840.previews.kube001.services.stellar-ops.com

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr840.previews.kube001.services.stellar-ops.com

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr840.previews.kube001.services.stellar-ops.com

stellar-jenkins commented:
**Something went wrong with PR preview build please check**

BlaineHeffron commented:
@janewang I opened a draft PR for a Cookbook.md file. https://github.com/stellar/stellar-cli/pull/1541  I also added to the script an entry to copy over this Cookbook.md file to `stellar-cli-cookbook.mdx` alongside the `stellar-cli.mdx` file. I also added a link to it in the docs. I am wondering if you would like to copy all the existing `stellar-cli` guides that are currently here? https://developers.stellar.org/docs/build/guides/cli

stellar-jenkins commented:
**Something went wrong with PR preview build please check**

stellar-jenkins commented:
**Something went wrong with PR preview build please check**

janewang commented:
Hey @BlaineHeffron The other file that's being copied over is this file: https://github.com/stellar/stellar-cli/blob/main/FULL_HELP_DOCS.md. I think we only need these two files to be copied.

stellar-jenkins commented:
**Something went wrong with PR preview build please check**

stellar-jenkins commented:
**Something went wrong with PR preview build please check**

BlaineHeffron commented:
This PR can only be merged after https://github.com/stellar/stellar-cli/pull/1541 is merged. In that PR I moved the guide mdx files to the CLI repo in a new  /docs directory. In this PR I altered the script to copy over the mdx files from the /docs directory.

stellar-jenkins commented:
**Something went wrong with PR preview build please check**

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr840.previews.kube001.services.stellar-ops.com

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr840.previews.kube001.services.stellar-ops.com

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr840.previews.kube001.services.stellar-ops.com

janewang commented:
You can run `yarn check:mdx` to fix the markdown linting error

BlaineHeffron commented:
The problem is that I needed to add the `yarn stellar-cli:build` command to the git actions so that markdown files that link to the CLI cookbook entries wouldn't error during build. This means that the CLI docs markdown file gets linted and apparently it is producing warnings. This is the auto generated docs file from the CLI. Either we would have to fix the markdown formatting errors from the generated document or we just add stub files for all of the CLI cookbook entries and don't actually run `yarn stellar-cli:build` in the git actions.

stellar-jenkins commented:
**Something went wrong with PR preview build please check**

stellar-jenkins commented:
**Preview is available here:**<br>http://developers-pr840.previews.kube001.services.stellar-ops.com

ElliotFriend commented:
@BlaineHeffron @janewang

Hi, friends!!

Sorry, i'm late to the party here, had a couple hectic weeks. Can I get a little more context about what brought this change about? I think I've picked up on reasonably on _what_ is taking place in this change, but I'm a little unclear about the _why_. Is there a particular reason having these cookbook guides live alongside the CLI codebase is better off than having them live in the docs?

One of the reasons I'm asking is: now we're seeing the same broken link error Blaine mentioned [in this comment](https://github.com/stellar/stellar-docs/pull/840#issuecomment-2341917745) if we're just using `yarn start` to preview locally while editing.

Thanks!

briwylde08 commented:
What @ElliotFriend said! I'm getting broken links when linking to the CLI guides and I'm not sure how to rectify the link formatting. We can't merge any other docs PRs until we fix these links.

janewang commented:
There is a test that runs through the command to make sure that they are valid on the CLI side. That test could only reasonably live in the CLI repo. That's why Blaine created the file on the CLI side and copied over to the docs. 

Can someone point me to the broken links on the docs side? I heard it mentioned but didn't see the links.

ElliotFriend commented:
@janewang you can see some of the fixes Bri was needing in some of the changed files in pr #982

janewang commented:
Thank you @briwylde08!
